tasks/gulp/compile-assets.mjs

```diff
@@ -83,7 +84,23 @@ compileStylesheets.displayName = 'compile:scss'
 function compileStylesheet (stream, options = {}) {
   return stream
     .pipe(plumber())
-    .pipe(sass(options))
+
+    // Compile Sass to CSS
+    .pipe(sass(options).on('error', (cause) => {
+      const error = new PluginError('gulp-sass', cause.messageFormatted)
+      error.showProperties = false
+
+      // Gulp continue (watch)
+      if (isDev) {
+        console.error(error.toString())
+        return stream.emit('end')
+      }
+
+      // Gulp exit with error
+      return stream.emit('error', error)
```

Comment:
**nitpick** I see what this does, but it feels a bit narrow. If we'd introduce a new plugin or custom transform that'd emit error, we'd be back to square one.

Wondering if something [like adding a final handling of `error` on the `watch` stream](https://github.com/specious/specious.github.io/commit/f8571d28#diff-25789e3ba4c2adf4a68996260eb693a441b4a834c38b76167a120f0b51b969f7R72) ([found via this StackOverflow comment](https://stackoverflow.com/questions/23971388/prevent-errors-from-breaking-crashing-gulp-watch#comment83646234_23973536)) could get us not to worry about task errors in dev long term?

Reply:
Ah sorry @romaricpascal I've only taken this from the docs:
https://github.com/dlmanning/gulp-sass#render-your-css

Their approach suggests we swallow all errors with [`this.emit('end')` inside the logger](https://github.com/dlmanning/gulp-sass/blob/c04bb67043fdbcb930152232ee9ba4caff0b2599/index.js#L174):

```js
.pipe(sass().on('error', sass.logError))
```

This is perfect for development and doesn't stop `gulp.watch()`
This is awful for production as we no longer "break the build"

So I've gone one step better and emitted an error when not developing

>Wondering if something [like adding a final handling of error on the watch stream](https://github.com/specious/specious.github.io/commit/f8571d28#diff-25789e3ba4c2adf4a68996260eb693a441b4a834c38b76167a120f0b51b969f7R72) ([found via this StackOverflow comment](https://stackoverflow.com/questions/23971388/prevent-errors-from-breaking-crashing-gulp-watch#comment83646234_23973536)) could get us not to worry about task errors in dev long term? It may be an outdated solution though 😬

We'd lose Gulp's `PluginError()` flagging **gulp-sass** as the culprit (with fancy error output). Plus we'd lose per-plugin control of `error.showProperties = false` where we know the log formatter already includes this

Also if you use `stream.on('error')` to trigger `stream.emit('error', error)` we get an infinite loop

Reply:
Thanks for the pointer to the `gulp-sass` recommendation 😄 
 
I think what I was looking for is a more systematic approach to handling transforms breaking, as for now it happens with Sass, but could happen on other transforms and the `watch` task would break again. It felt very gulp-sass-specific, hence my link to the example that turns `error` into `end` (not `error` so no looping) globally on the watch streams (so the build still breaks when not watching).

Similarly, the `PluginError` bit should ideally be part of any piping, imho. This way it won't just be Sass letting us know that it's gulp-sass breaking (though the error and stacktrace will likely already provide plenty to pinpoint things, no?)

If we're happy to revisit if a newer plugin breaks, then fine going with it.

Reply:
No worries, yeah I understand, it's good feedback

I've pushed again with the error handler moved into `gulp-plumber` so it listens for PostCSS errors too

>Similarly, the PluginError bit should ideally be part of any piping, imho. This way it won't just be Sass letting us know that it's gulp-sass breaking (though the error and stacktrace will likely already provide plenty to pinpoint things, no?)

Sounds good, I'll nominate you for taking this further 😆

Hopefully we're good to get this first improvement PR in though?

Reply:
Neat! Yeah, using `plumber` to handle makes things more generic 🙌🏻  Shame Gulp doesn't have that built in 😢